src/libs/Middleware/SaveResponseInOnyx.js

```diff
@@ -16,19 +16,12 @@ const requestsToIgnoreLastUpdateID = ['OpenApp', 'ReconnectApp', 'GetMissingOnyx
 function SaveResponseInOnyx(requestResponse, request) {
     return requestResponse.then((response) => {
         // Make sure we have response data (i.e. response isn't a promise being passed down to us by a failed retry request and response undefined)
-        if (!response) {
+        if (!response && !request.successData && !request.failureData) {
```

Comment:
**thought:** maybe we should default to an empty object for the response

We can then avoid most of the code changes due to the `response` now being possibly `undefined`.

Reply:
Yea, I tried that at first and it broke a bunch of tests. So I think using an empty response would require more code changes and be more involved than just simply returning an empty object

Reply:
Sounds like you agree the approach would be better though 😄. Should we fix or remove them if the tests are wrong?

Reply:
I remember most tests breaking with a default empty object, but I'd have to look at it again

Reply:
Nvm, it seems like all tests are passing when using an empty object as default. Not sure what changed since the last time I looked into this.